### ISSUE TITLE
FINAL VERDICT ON MACBOOK SUPPORT.(ALSO FOR VOLUNTEERS)

### ISSUE THREAD

**Issue by skulldealer** [ventoy/Ventoy#584]
TOPIC-
my views on macbook support for ventoy and further improvements for ventoy 2.0 beta release

-----------------------------------------------------------------------------------------   

GENERAL NOTE -
this thread is meant to address all the developers of ventoy and developers of any other github project which ventoy uses directly or indirectly.
for your personal views over this topic please quote reply to this post and let me know.

------------------------------------------------------------------------------------------

LEGAL NOTE - 
AS LONG AS A GITHUB PROJECT IS PUBLIC AND HAS A SOURCE CODE AVAILABLE TO THE PUBLIC, THAT CODE CAN BE COPIED, REDESIGNED AND REIMPLEMETED IN OTHER GITHUB PROJECTS.

------------------------------------------------------------------------------------------

ISSUE-
recently there has be a dissatisfaction in the community of macbook users towards ventoy. they are unable to boot into their desired iso files due to this limitation. i will not call this a bug because ventoy was not specifically developed by keeping macbook users in mind. I suppose all the compatiblity checks done by ventoy developers were only on windows laptop. 

------------------------------------------------------------------------------------------

MY VIEW ON THE ISSUE-
this issue has been unadressed for atleat 5 months as per my knowledge and i am finally adressing it with a complete solution.
this solution has not been extensively tested but shows promising results and holds strong potential to resolve the macbook's ventoy incompatibilities and also provide performance improvements to the overall project.

------------------------------------------------------------------------------------------

WHAT I BELIEVE-
making any change to the ventoy version 1.x.xx release cycle to forcefully provide support for macbooks looks like a bad idea as it probably would break ventoy's core components and creating incompatibility for windows users as well. This is a huge change in ventoy and will probably be fully ready with great stability after several releases of its beta cycle after macbook user opinion.

------------------------------------------------------------------------------------------

MY ADVICE TO VENTOY DEVELOPERS-
i suggest the ventoy developers to continue ventoy version one release and create ventoy version 2 beta release with experimental support for macbook users. once the macbook users are satisfied with the beta version 2 lineup and the beta version 2 lineup does not create any trouble for windows users, the first non-beta ( meaning stable release with macbook support)  version for ventoy 2.0 lineup can be released with full support for both macbook and windows users as well as many improvements.

------------------------------------------------------------------------------------------

SOME BASIC INFORMATION-
here are the changes that need to be done as per my research to enable support for macbook. this method will most likely have no effect on regular windows users. but the developer of ventoy need to reprogram the plugins already existing in ventoy to work with the newer version 2 release of ventoy.

here are the links of two intresting bootable usb projects on github called "unetbootin" and "multiboot usb".

1. UNETBOOTIN-
link - https://unetbootin.github.io/
link - https://github.com/unetbootin/unetbootin

there is all the information on how this tool works on the website.

known limitations with unetbootin- 
looks like unetbootin does not support arch linux, the reason of this is provided on blackarch linux official website. it says that and i quote, -
 
Do not use UNetBootIn to write ISO files to flashdrives. UNetBootIn modifies the bootloader configuration, which is bad. You can use this instead (where /dev/sdX is your flashdrive and file.iso is a BlackArch ISO)
(please refer to blackarch linux downloads page) 
Thats all i know about it so far.

2. multiboot usb-
link - http://multibootusb.org/
link - https://github.com/mbusb/multibootusb

there is all the information on how the tool works on the website

known limititations- nothing so far. 

why should ventoy use multiboot as well as unetbootin?
answer- incompatiblities created by unetbootin can be easily resolved by multiboot without messing up much with the UNETBOOTIN orignal code.

------------------------------------------------------------------------------------------

THE DETAILED PLAN-

NOTE - i am only providing a map or a blueprint whatever you may say. the real work for this have to be done by ventoy developer in reprogramming ventoy.

OK NOW UNDERSTAND THIS CAREFULLY.

ventoy already has an option to reserve free space on the usb it is installed on.

i have found that these two tools UNETBOOTIN and multiboot usb are capable of making a partition bootable.
so mbr disk can have a maximum of 4 primary partitions. after the fourth primary partition, logical partitions can be created-these are not bootable.

so there are two partitions used by ventoy. so two more primary partitions can be created.

that means if i reserve some space at the end of a usb while installing ventoy, that space can be formated into 2 primary partitions.

and these two tools can make these individual partitions bootable!
and these bootable partition may work for macbook users. also in case an iso is not working with ventoy, these two partition can be used as a rescue partition. so if ventoy fails to boot any iso on any device in general, these two tools can be used to boot thoose iso files. they will surely do it. this means that if ventoy uses UNETBOOTIN and muliboot-iso tools in its source code, ventoy will have a iso booting successs rate of hundred percent. if any iso fails to boot with ventoy's default method, then ventoy will create a log file (will talk about it later) and on updating ventoy these iso incompatible list of log will be sent to the developers and can fix them for next update. this will speed up ventoy development.

------------------------------------------------------------------------------------------

NOW HOW TO GET THIS TO WORK WITH VENTOY?

UNetbootin generates an appropriate syslinux config file in /syslinux.cfg, and makes USB drive partition bootable using syslinux. On Linux, if extlinux is installed and the target USB drive is ext2 or ext3, extlinux (already on ventoy) is used instead; the config file is installed in /extlinux.conf. The partition to which it has been installed is also marked as active.

this process needs to be executed in grub. you first need to find a way to extract the iso. i have seen that ventoy uses 7zip or some kind of compression and decompression utility. So then the ventoy developers can create a program which creates the required files like syslinux and others, to make the iso bootable. believe me unetbootin quickly flashes iso to without formatting them before. so data already stored on usb remains untouched. so iso flashing would not take as long as in rufus or balena etcher. 

but since this method is not guaranteed to boot all the iso files we use multiboot-iso tool.
please see on the website of the tool to know how it works.

------------------------------------------------------------------------------------------

WHAT DO WE NEED TO DO?

add a default injection plugin called ventoy-helper plugin. ( you can name it anything. )
it will store logs of ventoy in the 32 mb vtoyuefi partition by creating a vtoylog.txt
if ventoy fails to boot using default method which we are currently using it can be found from the text file vtoylog.txt
so if it failed to boot an iso, then it is logged in, if it successfully boots then the log is automatically cleared.( good for privacy and storage. )
if booting fails memdisk mode is activated on next time if user selects the same iso file as ventoy on every boot reads the vtoylog.txt file and if it finds that the same iso previously failed by the default method, it auto enables the memdisk mode for the same iso file.
if it fails again unetbootin mode is activated next time for the same iso which failed to boot by default and memdisk mode.
if it fails again multiboot iso is activated next time.

if all these fail a log file is created on usb called reportmelog.txt which is to be sent to ventoy developers on github or is automatically sent to them on updating ventoy.

if ventoy fails to boot after the first try, and it successfully boots into iso using any other method, it creates a profile and stores it with extension .prof . on next boot user can boot iso from that profile and skip wasting time by trial and error method. these profiles will just be ventoy.json kind of files but for seperate iso files. could be renamed.

i hope this should work.
before adding this the ventoy developer has to merge the code of UNETBOOTIN and multiboot-iso.

once all this has been done, the plugins need to reorogrammed to work with our new method.

all this cannot be done by a single release so you could release beta for ventoy 2 and list all missing features in it.

the grub project that ventoy uses also needs to be extensively reprogrammed specially for ventoy so it is able to do the tasks unetbootin and multiboot-iso do seperately.

the last thing you should do is add support for other languages.

i hope that helps.

------------------------------------------------------------------------------------------

MY PERSONAL SUGGESTIONS -

add the ventoy installer iso file by default to the ventoy usb and add xfce or lxqt desktop enviournment for them. will be great. i manually need to move the new iso installer every time i update the ventoy usb.
add a storage mode option in ventoy installer. it will create fifth logical partition called storage. files in it can be accessed during a live usb session. most changes have to be done in the partition configuration window of ventoy.

------------------------------------------------------------------------------------------

ANOTHER IMPORTANT SUGGESTION - (voulneteers may follow this step and let us know if it is working or not.)

first install ventoy on usb and preserve some space at the end of the usb.
then create a new primary partition and choose an iso file and flash it into that new primary partition by unetbootin.(partition has to be fat32)
then create another primary fat32 partition install multiboot iso tool into it. from it flash iso you want.
then create an exfat logical partition and rename it storage. youb can store files here and they can be accessed anytime from any booting method.

NOTE- any problems in partitiioning please use gparted. You can use another live usb and install ubuntu in it and use it to partition the drive.(let me know if i should post a step by step detailed guide for voulnteers.)

FOR VENTOY DEVELOPER-
then instead of trying iso in the new partitions, try them with ventoy. Debian based distros are unstable. example kali linux v 2020.3 boots into grub while some other debian iso dont.
this issue has to be solved before doing anything else.

------------------------------------------------------------------------------------------

MY OBSERVATIONS AFTER DOING THIS-

initial research shows this may improve ventoys support for hard drives also.
expecting that this update may change size of ventoy.
boot up time with mac support is little longer than usual. But success rate so far in booting looks like 100 percent
it will be an extension to memdisk mode. If any iso fails to boot from ventoy directly use memdisk first else use these new modes
few iso like tails seem to perform way better and boot way faster with newer methods as compared to ventoy.

here is a list of all the iso files i tried and which all worked with unetbootin inside ventoy

elementaryos-5.1-stable.20200814.iso
gparted-live-1.1.0-5-amd64.iso
kali-linux-2020.3-live-amd64.isohkubuntu-20.04.1-desktop-amd64.iso
linuxmint-20-cinnamon-64bit.iso
MX-19.2_September_x64.iso
neon-user-20201015-0947.iso
Parrot-security-4.10_amd64.iso
tails-amd64-4.10.iso.
ubuntu-20.04.1-desktop-amd64.iso
ubuntu-budgie-20.04.1-desktop-amd64.iso
ubuntudde-20.04.1-amd64-desktop.iso
Windows.iso
xubuntu-20.04.1-desktop-amd64.iso
Zorin-OS-15.3-Core-64-bit.iso
Zorin-OS-15.3-Lite-64-bit.iso

------------------------------------------------------------------------------------------

NOTE-

we will have to make sure it works cross models for macbooks as well as windows. please check and report bugs in this post only. do not post these bugs under supported/unsupported iso thread.
if this thread dies due to lack of support or any other reason, this post will always be available to restart the macbook support development again. hence i have made sure i provide my best work. 

------------------------------------------------------------------------------------------

Thank you.

**Comment by skulldealer** [comment]
I will be uploading a translation for same post again in simplified chinese
我会再次上传简化中文的同一帖子的翻译

**Comment by skulldealer** [comment]
TRANSLATION IN SIMPLIFIED CHINESE -

如果你正在读这篇文章， 那么你可能正在阅读翻译的简体中文 -
我用谷歌翻译。

注意 - 我不擅长翻译长文档。如果某事被重复或误解， 请自己重新翻译这个对齐的帖子


主题 -
我对 macbook 支持文托和文托 二 测试版的进一步改进的看法

-----------------------------------------------------------------------------------------   

一般注 -
此线程旨在解决 ventoy 的所有开发人员和任何其他 github 项目的开发人员， ventoy 直接或间接使用。
对于您个人对这个话题的看法， 请引用回复这篇文章， 让我知道。


------------------------------------------------------------------------------------------

法律说明 -
只要 GITHUB 项目是公开的，并且具有可供公众的源代码，该代码就可以在其他 GITHUB 项目中复制、重新设计和重新复制。

------------------------------------------------------------------------------------------

问题 -
最近， 在社区里， macbook 用户对文托伊感到不满。由于此限制，他们无法引导到所需的 iso 文件。我不会称之为错误， 因为 ventoy 不是专门开发的， 记住 macbook 用户。我想所有由 ventoy 开发人员完成的可质量检查都只在 Windows 笔记本电脑上。

------------------------------------------------------------------------------------------

我对这个问题的看法
根据我的知识， 这个问题已经 5 个月了， 我终于用一个完整的解决方案来解决它。
此解决方案尚未经过广泛测试，但显示出有希望的结果，并具有强大的潜力，可以解决 macbook 的通风不兼容问题，并为整个项目提供性能改进。

------------------------------------------------------------------------------------------

我所相信的
对 ventoy 版本 1.x.xx 发布周期进行任何更改，以强制支持 macbook，看起来是个坏主意，因为它可能会破坏 ventoy 的核心组件，并为窗口用户造成不兼容。这是一个巨大的变化，在文托，并可能完全准备好与伟大的稳定性后，其测试版周期的几个版本后，macbook用户的意见。

------------------------------------------------------------------------------------------

我给文托伊开发人员的建议 -
我建议文托伊开发人员继续 ventoy 版本一个版本， 并创建 ventoy 版本 2 测试版与 macbook 用户的实验支持。一旦 macbook 用户满意测试版 2 阵容和测试版 2 阵容不给窗口用户带来任何麻烦， 第一个非测试版 （意味着稳定发布与 macbook 支持） 版本为 ventoy 2.0 阵容可以发布与全面支持 macbook 和窗口用户以及许多改进.

------------------------------------------------------------------------------------------

一些基本信息 -
以下是需要根据我的研究进行的更改， 以便支持 macbook 。此方法很可能对常规窗口用户没有任何影响。但 ventoy 的开发人员需要重新编程已经存在的文托伊插件， 以与 ventoy 的较新版本 2 版本工作。

这里是 github 上两个可启动 usb 项目的链接， 称为 "unetbootin" 和 "多引导 usb" 。

1. 乌内布汀 -
链接 - https://unetbootin.github.io/
链接 - https://github.com/unetbootin/unetbootin

有关于此工具如何在网站上工作的所有信息。

已知限制与 unetbootin -
看起来 unetbootin 不支持拱 linux， 这是在 Blackarch linux 官方网站上提供的原因。它说， 我引用， -

不要使用 UNetBootin 将 ISO 文件写入闪存驱动器。UNetBootIn 修改引导加载程序配置，这是不好的。您可以改为使用此选项（其中 /dev/sdX 是您的闪存驱动器，文件.iso 是 Blackarch ISO）
（请参阅黑淀粉 linux 下载页面）
这就是我到目前为止对它了解的一切。

2. 多控制器 usb -
链接 - http://multibootusb.org/
链接 - https://github.com/mbusb/multibootusb

有关于该工具如何在网站上工作的所有信息

已知的限制 - 到目前为止什么都没有。

为什么文托伊应该使用多靴和乌内博汀？
答案 - 由 unetbootin 创建的不兼容可以通过多引导轻松解决， 而不会与 Unetbootin 或ignal 代码混为一提。

------------------------------------------------------------------------------------------

详细计划-

注意 - 我只是提供地图或蓝图， 无论你说什么。真正的工作必须由文托开发人员在重新编程文托伊。

好了， 现在仔细理解这一点。

文托伊已经有一个选项， 以保留可用空间的 usb， 它安装在它。

我发现，这两个工具UNETBOOTIN和多引导usb能够使分区启动。
因此，mbr 磁盘最多可以有 4 个主分区。在第四个主分区之后，可以创建逻辑分区-这些分区不可启动。

所以文托伊有两个分区。以便可以创建两个主分区。

这意味着， 如果我在安装 ventoy 时在 usb 的末尾保留一些空间， 该空间可以格式化为 2 个主分区。

这两个工具可以使这些单独的分区可启动！
这些可引导分区可能适合 macbook 用户。也万一一一个等值不工作的文托，这两个分区可以用作救援分区。因此，如果 ventoy 一般无法在任何设备上启动任何 iso，这两个工具可用于启动 thoose iso 文件。他们一定会做的。这意味着，如果 ventoy 在其源代码中使用 UNETBOOTIN 和 muliboot-iso 工具，则 ventoy 的 iso 引导成功率将为 100%。如果任何 iso 未能使用 ventoy 的默认方法启动，则 ventoy 将创建一个日志文件（稍后将讨论），在更新 ventoy 时，这些 iso 不兼容的日志列表将发送给开发人员，并可以修复它们以进行下一次更新。这将加快文托的发展。

------------------------------------------------------------------------------------------

现在如何让这个与文托工作？

UNetbootin 在 /syslinux.cfg 中生成适当的 syslinux 配置文件，并使用 syslinux 使 USB 驱动器分区可启动。在 Linux 上，如果安装了 extlinux，并且目标 USB 驱动器为 ext2 或 ext3，则使用 extlinux（已在 ventoy 上）;如果已安装 extlinux，则使用 extlinux（已在 ventoy 上）;配置文件安装在 /extlinux.conf 中。已安装到的分区也标记为活动。

此过程需要在 grub 中执行。您首先需要找到一种方法来提取 iso。我见过， 文托伊使用 7zip 或某种压缩和减压实用程序。因此，ventoy 开发人员可以创建一个程序，该程序可以创建所需的文件，如 syslinux 和其他文件，使 iso 可引导。相信我 unetbootin 迅速闪烁 iso 到没有格式化他们之前。因此，已存储在 usb 上的数据保持不变。所以等值闪烁不会花那么长的时间在鲁弗斯或巴莱纳等。

但由于此方法不能保证启动所有等值文件，我们使用多引导-iso工具。
请参阅该工具的网站，了解其工作原理。


------------------------------------------------------------------------------------------

我们需要做什么？

添加一个默认的注射插件称为文托伊帮助插件。（你可以把它命名任何东西。
它将通过创建 vtoylog. txt 将 ventoy 的日志存储在 32 mb vtoyuefi 分区中
如果 ventoy 未能使用默认方法启动， 我们目前正在使用它， 可以从文本文件 vtoylog. txt 找到
因此，如果失败启动 iso，则它登录，如果它成功启动，则日志将自动清除。（适合隐私和存储。
如果用户在每次启动时选择与 ventoy 相同的 iso 文件，并且发现默认方法以前失败的相同 iso，则下次激活了 memdisk 模式，则它自动启用同一 iso 文件的 memdisk 模式。
如果再次失败， unetbootin 模式将激活下一次相同的 iso， 失败启动默认情况下和 memdisk 模式。
如果再次失败， 下次将激活多引导 iso 。

如果所有这些失败，将在 usb 上创建一个日志文件，称为 reportmelog.txt，该文件将发送到 github 上的文通开发人员，或在更新 ventoy 时自动发送给他们。

如果 ventoy 在第一次尝试后无法启动，并且它使用任何其他方法成功引导到 iso，它将创建一个配置文件并存储它与扩展 .prof 。在下次启动时， 用户可以从该配置文件启动 iso， 并跳过通过试验和错误方法浪费时间。这些配置文件将只是 ventoy. json 类的文件， 但对于单独的 iso 文件。可以重命名。

我希望这应该行得通。
在添加此之前， ventoy 开发人员必须合并 UNETBOOTIN 和多引导等值的代码。

一旦这一切完成，插件需要重新编程，以工作与我们的新方法。

所有这一切不能由单个版本完成， 因此您可以发布 ventoy 2 的测试版， 并列出它中所有缺少的功能。

文托伊使用的垃圾项目还需要对文托伊进行广泛重新编程， 以便它能够单独完成 unetbootin 和多引导 - iso 的任务。

您最不应该做的事情是添加对其他语言的支持。

我希望这有帮助。

------------------------------------------------------------------------------------------

我个人建议 -

默认情况下将 ventoy 安装程序 iso 文件添加到 ventoy usb， 并添加 xfce 或 lxqt 桌面环境。会很棒我手动需要移动新的 iso 安装程序， 每次我更新文托 usb 。
在文托安装器中添加存储模式选项。它将创建称为存储的第五个逻辑分区。在实时 usb 会话期间可以访问中的文件。大多数更改必须在 ventoy 的分区配置窗口中完成。

------------------------------------------------------------------------------------------

另一个重要建议 - （破坏者可能会按照此步骤， 让我们知道它是否工作。

首先在 usb 上安装通风口， 并在 usb 的末尾保留一些空间。
然后创建一个新的主分区并选择一个等值文件，然后由 unetbootin 将它闪存到新的主分区中。（分区必须脂肪32）
然后创建另一个主 fat32 分区安装多引导等值工具到它。从它闪光 iso 你想要的。
然后创建 exfat 逻辑分区并重命名其存储。youb 可以在这里存储文件， 可以随时从任何引导方法访问它们。

注意 - 在分手时出现任何问题，请使用 gparted。您可以使用另一个实时 usb 并在它中安装 ubuntu，并用它来分区驱动器。（让我知道， 如果我应该张贴一步一步的详细指南， 为吸血鬼。

对于文托伊开发人员 -
然后， 而不是尝试 iso 在新的分区， 尝试他们与 ventoy 。基于 Debian 的分张是不稳定的。例如 kali linux v 2020.3 引导到 grub， 而一些其他 debian iso 不。
这个问题必须解决之前，做任何事情。

------------------------------------------------------------------------------------------

做我的观察后 -

初步研究表明，这可能改善硬盘的通风支持。
期待此更新可能会改变文托的大小。
使用 mac 支持的启动时间比平常长一点。但到目前为止， 在引导中的成功率看起来像 100%
这将是对 memdisk 模式的扩展。如果任何 iso 无法从 ventoy 启动，请先使用 memdisk，然后使用这些新模式
很少像尾巴这样的 iso 似乎表现更好， 与 ventoy 相比， 用较新的方法启动方式更快。

这里是一个列表的所有 iso 文件， 我尝试， 所有与 unetbootin 内文托工作

基本 - 5.1 - 稳定. 20200814. iso
gparted - live - 1.1.0 - 5 - amd64. iso
kali - linux - 2020.3 - live - amd64. isohkubuntu - 20. 04. 1 - 桌面 - amd64. iso
linuxmint - 20 - 肉桂 - 64 位. iso
MX-19.2_September_x64.iso
霓虹灯用户-20201015-0947.iso
鹦鹉安全- 4.10_amd64. iso
尾巴 - amd64 - 4.10. iso 。
ubuntu-20.04.1-桌面-amd64.iso
ubuntu - budgie - 20.04. 1 桌面 - amd64. iso
ubuntudde - 20.04. 1 - amd64 桌面. iso
视窗. iso
xubuntu-20.04.1-桌面-amd64.iso
佐林-OS-15.3-核心-64位.iso
佐林-OS-15.3-Lite-64 位.iso

------------------------------------------------------------------------------------------

注 -

我们必须确保它的工作原理交叉模型的 macbook 以及窗口。请仅在此帖子中检查和报告错误。不要在受支持/不支持的等值线程下发布这些 Bug。
如果此线程因缺乏支持或任何其他原因而死亡， 此帖子将始终可用于重新启动 macbook 支持开发。因此， 我已经确保我提供我最好的工作。

------------------------------------------------------------------------------------------

谢谢。

**Comment by skulldealer** [comment]
sorry for the delay and thank you for waiting.
很抱歉耽误了时间， 谢谢你的等待。

**Comment by skulldealer** [comment]
Voulnteers will be testing only popular and mainstream distros. After then we shall work to add support for other distros

Distros voulnteers will be testing (Only Debian in the beginning)

Ubuntu(if works means same for all flavours)
Mint
Kali
Parrot
Tails OS

If you have suggestions please drop them down here. May edit this post and add them.

At the beginning we look forward to ensure support for major distros. Kindly dont ask for testing distros like samsung ssd firmware or any old distros that are nomore supported. We will first add support for mainstream distros for debian linux in phase 1 and in phase 2 we will work with windows 2h2 iso. In last phase we will work for arch support.

After then only support for other distros will be added.

Time of voulnteers is precious so kindly understand.

Voulnteers are allowed to test any distro they want after tesing distros we tell. After testing, the voulnteers will have to submit their experience and screenshot of the distro running on macos. 

Instructions to be followed by them are give in the main orignal post. Kindly check there.

Requirements for voulnteers-
A macbook
A windows pc
A 16 gb usb atleast 
Familiarity with bootcamp on mac.

**Comment by skulldealer** [comment]
Voulnteers are requested to message me in this thread itself.

Do not reply on older threads.

Read everything in the post before replying.

In case if voulnteers are facing any issue with the process I have posted in the major thread, they may feel free to contact me. If you want I can provide a more detailed step by step guide to the voulnteers.

Thank you.

**Comment by julianpoemp** [comment]
>if i reserve some space at the end of a usb while installing ventoy, that space can be formated into 2 primary partitions.
>
> and these two tools can make these individual partitions bootable!

I did that with Kali on my USB stick to have a system with persistence:

1. Install ventoy and do the option to preserve data at the end of the usb stick.
2.  Use a disk partition tool and split the last partition of the USB stick into two pieces: 1) Partition for kali system FAT32, 2) Partition ext4 with label "persistence".
3. Use unebootin to write the ISO to the first partition (third partition over all).
4.  On the second partition (fourth partition over all) create a file called "persistence.conf" and add content "/ union".

Unfortunately my solution is still a workaround. I hope ventoy fixes this problem so that no workaround with third party software is needed.

**Comment by skulldealer** [comment]
> > if i reserve some space at the end of a usb while installing ventoy, that space can be formated into 2 primary partitions.
> > and these two tools can make these individual partitions bootable!
> 
> I did that with Kali on my USB stick to have a system with persistence:
> 
> 1. Install ventoy and do the option to preserve data at the end of the usb stick.
> 2. Use a disk partition tool and split the last partition of the USB stick into two pieces: 1) Partition for kali system FAT32, 2) Partition ext4 with label "persistence".
> 3. Use unebootin to write the ISO to the first partition (third partition over all).
> 4. On the second partition (fourth partition over all) create a file called "persistence.conf" and add content "/ union".
> 
> Unfortunately my solution is still a workaround. I hope ventoy fixes this problem so that no workaround with third party software is needed.

Thanks a lot.
That's the thing I am willing to explain by my research. Unetbootin uses 7zip to unpack iso to the partition and then uses a algorithm for making that partition bootable. This algorithm is called syslinux. Ventoy already has 7zip utility built into the 32 mb vtoyefi partition. If it uses that algorithm it could simply be mean support for mac. Mac works well with syslinux. Ventoy needs to figure out a way to use syslinux in ventoy. Then Mac's will be supported. Also many unsupported iso files have complete support for syslinux. Which is great.

In short I am saying that what unetbootin does is not complicated. It can be easily replicated by ventoy. The efficiency of the replication decides the booting time for mac users.

**Comment by skulldealer** [comment]
Insted of using unetbootin, you can also use multibootiso tool. I have given link for it above. It does the same thing but is more reliable and has option to multiboot. Only restriction is only installable on windows.

**Comment by julianpoemp** [comment]
> Insted of using unetbootin, you can also use multibootiso tool. I have given link for it above. It does the same thing but is more reliable and has option to multiboot. Only restriction is only installable on windows.

I tried multiboot once, but this is not as easy as ventoy. Are you sure that the solution is using syslinux instead of changing the grup configuration? Perhaps the solution can be found if the ventoy developers compares their grub config with that one by unebootin. (I don't have any experience in programming Grub or using syslinux)

**Comment by skulldealer** [comment]
I have another suggestion for ventoy.

He could use unetbootin inside with every release.

**Comment by skulldealer** [comment]
I would like to know suggestions and opinion of ventoy over this topic.

**Comment by skulldealer** [comment]
> > Insted of using unetbootin, you can also use multibootiso tool. I have given link for it above. It does the same thing but is more reliable and has option to multiboot. Only restriction is only installable on windows.
> 
> I tried multiboot once, but this is not as easy as ventoy. Are you sure that the solution is using syslinux instead of changing the grub configuration? Perhaps the solution can be found if the ventoy developers compares their grub config with that one by unebootin. (I don't have any experience in programming Grub or using syslinux)

See, ventoy works completely different and unetbootin works completely different. You could say that by the fact that ventoy is easier to use. Tools like Rufus, Unetbootin use syslinux as bootloader and not grub. What I feel is if supprt if syslinux is added on ventoy, mac support could be ensured.

Please google syslinux yourself.

**Comment by skulldealer** [comment]
I am explaining the whole situation as simple as I could.

Ventoy is an airplane.
So it's easier and convinent rather than going all the way with a car.

But for some users airplane is not the solution everytime and they need a car sometimes.
So why not keep a car in the airplane?
This would not bother the users not interested in having a car but would be more convenient for users who need a car and earlier used third party to get the car.


Here,
Airplane is ventoy,
Car is unetbootin and specially syslinux,
Users who need car means mac users. 

You definitely don't need any programming skills to understand this.

Thank you.

**Comment by skulldealer** [comment]
I may not be able to get everything done by myself or be the only one participating in the thread. I want @ventoy and @a1ive to participate in the thread. The are excellent and the best people who could make this possible.
The knowledge of bootloaders, grub and moving files in bootloaders which @a1ive has is unmatched by anyone among us. These 2 are the right people who can get the job done.

**Comment by skulldealer** [comment]
With the latest release notes of ventoy 1.0.28, Ventoy has officially declined the proposal of using seperate bootloader in ventoy by quoting-

Recently, I received some special error reports, which finally proved to be the cause of the non-standard Ventoy environment. Especially use Ventoy in a single partition environment.

Although Ventoy displays Unofficial's label information, users will not care and will still think it is a Ventoy problem.

Ventoy is an overall design from the beginning, and it has not considered integration into other bootloader or partition environments.
Ventoy's verification and subsequent development of new features will not consider this non-standard usage.

Therefore, I decided to prohibit the use of Ventoy in a non-standard environment and no longer accept requests for deregulation.

Of course, this is just the default behavior of Ventoy. Ventoy is still 100% open source. If you want to apply Ventoy in your own environment, you can fork a branch and modify the source code implementation.

~Ventoy.







I feel sorry for the mac users and I don't see ventoy support for you all in the near future.
So I am closing the thread with the comment. In case anyone someone sees this in the future and has intrest in adding macbook support to ventoy may feel free to use my thread's information and create a new fork of ventoy for macbook users. All I want to say in short is using a supplementary bootloader along ventoy like isolinux or sys linux may be the solution. Anyways, A workaround is provided  in above threads for mac users which is not seamless but works.
Will hope the issue is resolved someday. Any new issue on macbook with ventoy should be closed with reference issue #584 as all possible queries to the compatibility issue is answered here.
Thank you.
Appreciate you all.
Peace.

**Comment by Rastafabisch** [comment]
I just came across Ventoy and loved the perspective it provides. As I am Mac user I tried to follow the Mac specific discussions, and I must say that I cannot really follow.
Macs are perfectly capable of booting with grub and even booting linux iso files from fat32 volumes. ([see this 6 year old post for reference](https://astrofloyd.wordpress.com/2014/01/14/boot-linux-from-usb-flash-drive-on-macbook/))
I already tried a Windows generated Ventoy USB today and while it did got stuck while Ubuntu 20.10 loaded it still did handed over booting (chain loading?) to the Ubuntu integrated grub loader. I will conduct some further testing and report back.
